packages/client/src/components/OrderDataCompletion/OrderDataCompletion.tsx

```diff
@@ -108,11 +112,78 @@ const OrderDataCompletion = () => {
     setFloor(addressObject.floor || '');
     setPostcode(addressObject.postcode);
     setCity(addressObject.city);
+    setOldStateAddress([
+      addressObject.street,
+      addressObject.streetNumber,
+      addressObject.apartmentNumber || '',
+      addressObject.floor || '',
+      addressObject.postcode,
+      addressObject.city,
+    ]);
+  };
+
+  const handleSelectDeliveryHours = (e: SelectChangeEvent<string>) => {
+    e.preventDefault();
+    setDeliveryHours(e.target.value);
+    setDeliveryHourStart(e.target.value);
+  };
+
+  const updateUserProfile = async (updateData: UpdateUserDto) => {
+    try {
+      await new UserssProfileApi(apiConfiguration).update({ updateUserDto: updateData });
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
+  };
+
+  const createUserAddress = async (updateData: CreateAddressDto) => {
+    try {
+      const getAddressApi = await new UsersAddressesApi(apiConfiguration).create({ createAddressDto: updateData });
+      setAddressId(getAddressApi.id);
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
   };
 
+  // check if user put new address
+  useEffect(() => {
+    const newAddressTable = [street, streetNumber, apartmentNumber, floor, postcode, city];
+    setAddress({ street, postcode, streetNumber, apartmentNumber, city });
+    if (JSON.stringify(oldStateAddress) === JSON.stringify(newAddressTable)) setStateChangedAddress(false);
+    else setStateChangedAddress(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Nie mam jak sprawdzić co tu konkretnie wywalało, ale możesz dodać komentarz uzasadniający dlaczego musiałeś tak zrobić bo to wygląda na code smell.

Reply:
cięzko dodać mały komentarz, ale coś dodam. Zamysł jest taki, że user wybiera adres z listy. Następnie klika sobie na textfield np z ulicą. Nic tam nie zmienia, jednakże w tym przypadku poleciałby request o stworzenie nowego adresu. W ten sposób zabezpieczamy się przed pustym klinkięciem w textfieldy.

Reply:
Najlepiej by było iterować po get z adresami, ale chyba dużo roboty z tym by bylo, aby kazde pole sprawdzac z kazdego adresu.

Reply:
Aj, miałem wrócić do tego fragmentu jak już będę przy kompie ale zapomniałem, jutro spojrzę i może wymyślę coś sensownego.

Reply:
Ok, sprawdziłem (troszkę później niż miałem, ale jak mówiłem na Discordzie musiałem od nowa stawiać wczoraj system :/), jedyne o co przyczepia się linter to `setAddress`, który pochodzi z `useState` (ale IDE tego nie widzi), a zgodnie z dokumentacją:
> React daje gwarancje, że funkcja `setState` jest tożsamościowa i że nie zmienia się podczas kolejnych renderowań.

Więc możesz je po prostu dopisać do tablicy i powinno działać tak samo (a nie powodować błędu więc `eslint-disable-next-line` staje się zbędne).